syntax = "proto3";

package vector_service;

service VectorService {
  rpc UploadFile(UploadFileRequest) returns (UploadFileResponse) {}
  rpc Search(SearchRequest) returns (SearchResponse) {}
  rpc Delete(DeleteRequest) returns (DeleteResponse) {}
}

message UploadFileRequest {
  bytes content = 1;
  string filename = 2;
}

message FileMetadata {
  string file_id = 1;
  string file_name = 2;
  int32 images_num = 3;
  int32 text_chunks_num = 4;
}

message UploadFileResponse {
  string message = 1;
  repeated FileMetadata result = 2;
}

message SearchRequest {
  string query = 1;
  int32 top_results = 2;
  string type = 3;
  repeated string file_ids = 4;
}

message SearchResult {
  string id = 1;
  string type = 2;
  float distance = 3;
  string content = 4;
  string file_id = 5;
  map<string, string> metadata = 6;
}

message SearchResponse { repeated SearchResult results = 1; }

message DeleteRequest { string id = 1; }

message DeleteResponse {
  string message = 1;
  bool success = 2;
}
